### DISCUSSION TITLE
2007/08/04/%E7%BC%93%E5%AD%98%E9%87%8D%E7%94%A8%E7%BB%93%E6%9E%84

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#175] [Announcements]
# 2007/08/04/%E7%BC%93%E5%AD%98%E9%87%8D%E7%94%A8%E7%BB%93%E6%9E%84

一、缓存重用

https://blog.qizhen.xyz/2007/08/04/%E7%BC%93%E5%AD%98%E9%87%8D%E7%94%A8%E7%BB%93%E6%9E%84

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2009年03月11日 17:02 编辑
传入移位寄存器时，还要看数据线有没有分叉，是从一个变量传过来的，还是从一个常量传过来的。如果是从变量传入，没有数据线分叉是不需拷贝的。移位寄存器的右边也有黑点是与你的后续程序有关的。如果把它连到一个可重用的节点上是不会有数据拷贝的。

回复
gufei
2009年03月10日 16:15 编辑
请问，使用移位寄存器的方法缓存重用时，数组传入寄存器的时候，labview有没有复制数据？仔细看，移位寄存器的右边也有黑点啊！

回复
arrowpig
2007年08月4日 18:35 编辑
 LabVIEW更新好快啊，我才离开NI两个礼拜就出来这么新的东西。现在LabVIEW对内存的控制越来越精确了。最好能够智能化的实现这样的优化才好。在C语言里面，所有的内存使用很多都归结到malloc/free。这使得设计一些高效的内存库成为可能。但是LabVIEW得内存是自动处理的。最后也能有个库来模块化内存就好了。
